include/seqan3/utility/type_list/traits.hpp

```diff
@@ -114,6 +114,33 @@ auto reverse(type_list<head_t, pack_t...>)
     return concat(reverse(type_list<pack_t...>{}), type_list<head_t>{});
 }
 
+//!\brief Constructs the set difference `list1 \ list2` [recursion anchor]
+template <typename ...elements1_t>
+constexpr seqan3::type_list<elements1_t...> type_list_difference(seqan3::type_list<elements1_t...>, seqan3::type_list<>)
+{ return {}; }
+
+//!\brief Constructs the set difference `list1 \ list2` [recursion]
+template <typename ...elements1_t, typename head2_t, typename ...elements2_t>
```

Comment:
```suggestion
template <typename ...current_list_t, typename remove_t, typename ...remove_list_t>
```

I think something along this line would make the code easier to understand?

Reply:
I didn't use list, as this is not a type list but a type pack